### DISCUSSION TITLE
Not smooth motion during very low speed

### DISCUSSION THREAD

**Discussion by kanta** [ponoor/step-series-support#7] [Q&A]
This thread is ported from this post;
https://github.com/ponoor/step-series-support/discussions/1#discussioncomment-2042000

Quote from the post of @florianillumination

> Hello,
> We are building a motorised slider for a camera and we have a small issue.
> We feel step during accelerating and decelerating. It is ok during normal speed. We are using our step motor very slow and we need a movement as smooth as possible and as quiet as possible.
> You can see the issue in the video attached.
> 
> Thank you in advance for your help.
> 
> Florian
> 
> We have a STEP400.
> 
> The setup is:
> /setKval 1 15 30 30 30
> /setSpeedProfile 1 5 5 30 , then 1 15 15 50 , then 1 20 20 100
> /run 1 100
> 
> INT_SPEED 1032
> ST_SLP 25
> FN_SLP_ACC 41
> FN_SLP_DEC 41
> 
> Power Supply: 24V - 13A
> 
> Motor information:
> https://www.omc-stepperonline.com/nema-23-bipolar-1-8deg-1-16nm-164-3oz-in-1-5a-5-4v-57x57x56mm-4-wires.html

https://user-images.githubusercontent.com/98390269/150960886-155013ac-9707-4029-8be2-8387096eb0fa.mp4

**Comment by kanta** [comment]
@florianillumination Thank you for choosing STEP400. 
There is a limitation of the smoothness of the stepper motor at extremely slow speed, but there is a possibility to improve it with parameter settings. Let me try with my environment, I don't have the exactly same motor but try with similar one.
Another option could be geared stepper motors, something like these; 
https://www.omc-stepperonline.com/precision-planetary-gearbox/

**Reply by kanta** [reply]
Hi again @florianillumination,
Probably higher KVAL settings could improve the motion.

I have tested with a NEMA23 motor rated as 3A/3.3ohm/2.9mH.

https://user-images.githubusercontent.com/1097902/151114371-f25fe044-5f6b-4479-9663-228ac75c9eb7.mp4


The initial KVAL setting is `/setKval 255 16 31 31 31`
Then test the motion with `/run 1 20.`. In this setting, the motion is obviously unstable.

Next, increase KVAL values. When the value reaches about 120, the movement seems to be the most stable. If you exceed that value, the vibration will increase again. With this setting, the acceleration and deceleration is much smoother than the initial condition.

Note : Higher KVAL settings means higher current consumption and the motor driver chip will be hotter. Please consider to use cooling fan and heat sink for the driver chip.

**Reply by florianillumination** [reply]
Excellent, Thank you very much for this demo. I will work on it and come back to you.

**Comment by florianillumination** [comment]
Hello @kanta 
So I tried to play with KVAL and at the end I played a bit too much with it because I burned the motor. I forget the maximum voltage of the motor.
Then I have to buy a new motor.
The thing that is different in my setup comparing yours is the weight or inertia. As you can see on the pictures attached, I need force at the beginning to make it moves. The weight of the moving plate (including the camera) is approx. 10kg.
What kind of motor could you advise me? Which power, voltage, etc. 
I don't need to be fast (maximum 10 secondes to go accross the slider), but I need to have a very smooth accelerating and decelerating. ;-)

Thank you in advance
Flo
![IMG_20220223_163307](https://user-images.githubusercontent.com/98390269/155622627-f3e17ea0-9f65-433b-8155-9366083f348e.jpg)
![IMG_20220223_163243](https://user-images.githubusercontent.com/98390269/155622603-16e0e167-62bc-4e73-8c79-fc665f7368ea.jpg)
![IMG_20220223_163252](https://user-images.githubusercontent.com/98390269/155622617-d54679af-d78c-4ad4-bc44-8c8483c6027c.jpg)
![IMG_20220223_163259](https://user-images.githubusercontent.com/98390269/155622619-ff159c88-5044-41f4-a8f1-087f86da91dd.jpg)

**Reply by kanta** [reply]
Ouch! I'm sorry to hear the motor was burnt. It sounds an unusual result, usually a motor driver chip might be burnt before a motor burnt. 
Anyway, I loved your professional camera jig setting!  Selecting a motor is complicated process, and very depended on small details of mechanism and desired motion. I know many motor manufacturers provide a motor selection tool but always difficult to fill complicated physical parameters. 
So some try and errors will be required to select the optimal one. What kind of motor are you using right now? And What kind of problems with it?
